infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/ExternalServerIngressBuilder.java

```diff
@@ -0,0 +1,125 @@
+import io.fabric8.kubernetes.api.model.extensions.IngressBackendBuilder;
+import io.fabric8.kubernetes.api.model.extensions.IngressRule;
+import io.fabric8.kubernetes.api.model.extensions.IngressRuleBuilder;
+import io.fabric8.kubernetes.api.model.extensions.IngressSpec;
+import io.fabric8.kubernetes.api.model.extensions.IngressSpecBuilder;
+import java.util.HashMap;
+import java.util.Map;
+import org.eclipse.che.api.core.model.workspace.config.ServerConfig;
+import org.eclipse.che.workspace.infrastructure.kubernetes.Annotations;
+
+/** @author Guy Daich */
+public class ExternalServerIngressBuilder {
+
+  private String host;
+  private String path;
+  private String name;
+  private String serviceName;
+  private IntOrString servicePort;
+  private Map<String, ? extends ServerConfig> serversConfigs;
+  private String machineName;
+  private Map<String, String> annotations;
+
+  public ExternalServerIngressBuilder withHost(String host) {
+    this.host = host;
+    return this;
+  }
+
+  public ExternalServerIngressBuilder withPath(String path) {
+    this.path = path;
+    return this;
+  }
+
+  public ExternalServerIngressBuilder withName(String name) {
+    this.name = name;
+    return this;
+  }
+
+  public ExternalServerIngressBuilder withServiceName(String serviceName) {
+    this.serviceName = serviceName;
+    return this;
+  }
+
+  public ExternalServerIngressBuilder withAnnotations(Map<String, String> annotations) {
+    this.annotations = annotations;
+    return this;
+  }
+
+  public ExternalServerIngressBuilder withServicePort(String targetPortName) {
+    this.servicePort = new IntOrString(targetPortName);
+    return this;
+  }
+
+  public ExternalServerIngressBuilder withServers(
+      Map<String, ? extends ServerConfig> serversConfigs) {
+    this.serversConfigs = serversConfigs;
+    return this;
+  }
+
+  public ExternalServerIngressBuilder withMachineName(String machineName) {
+    this.machineName = machineName;
+    return this;
+  }
+
+  public Ingress build() {
+
+    IngressBackend ingressBackend =
+        new IngressBackendBuilder()
+            .withServiceName(serviceName)
+            .withNewServicePort(servicePort.getStrVal())
+            .build();
+
+    HTTPIngressPathBuilder httpIngressPathBuilder =
+        new HTTPIngressPathBuilder().withBackend(ingressBackend);
+
+    if (!isNullOrEmpty(path)) {
+      httpIngressPathBuilder.withPath(path);
+    }
+
+    HTTPIngressPath httpIngressPath = httpIngressPathBuilder.build();
+
+    HTTPIngressRuleValue httpIngressRuleValue =
+        new HTTPIngressRuleValueBuilder().withPaths(httpIngressPath).build();
+    IngressRuleBuilder ingressRuleBuilder = new IngressRuleBuilder().withHttp(httpIngressRuleValue);
+
+    if (!isNullOrEmpty(host)) {
+      ingressRuleBuilder.withHost(host);
+    }
+
+    IngressRule ingressRule = ingressRuleBuilder.build();
+
+    IngressSpec ingressSpec = new IngressSpecBuilder().withRules(ingressRule).build();
```

Comment:
@guydaichs Would you be willing to contribute unit tests for this class?

Reply:
done